pkg/services/sqlstore/session.go

```diff
@@ -55,14 +48,24 @@ func startSessionOrUseExisting(ctx context.Context, engine *xorm.Engine, beginTr
 	}
 
 	newSess.Session = newSess.Session.Context(ctx)
+
 	return newSess, true, nil
 }
 
-// WithDbSession calls the callback with a session.
+// WithDbSession calls the callback with the session in the context (if exists).
+// Otherwise it creates a new one that is closed upon completion.
+// A session is stored in the context if sqlstore.InTransaction() has been been previously called with the same context (and it's not committed/rolledback yet).
 func (ss *SQLStore) WithDbSession(ctx context.Context, callback DBTransactionFunc) error {
 	return withDbSession(ctx, ss.engine, callback)
 }
 
+// WithDbSession calls the callback with a new session that is closed upon completion.
+func (ss *SQLStore) WithDbSessionForceNewSession(ctx context.Context, callback DBTransactionFunc) error {
```

Comment:
nit: this method name is a bit confusing to me. I don't think this name is perfect either, but maybe something closer to `ForceNewSession()`?

Reply:
WDYT about `WithNewDbSession()`?

Reply:
I like it as long you think it conveys the necessary information!